test/utils.h

```diff
@@ -395,12 +395,14 @@ inline bool maybeClearAllocator(int64_t max_bytes = ((int64_t)1 << 32)) {
 #if TORCH_VERSION_GREATER(2, 0, 1)
     // GetDevice was introduced in https://github.com/pytorch/pytorch/pull/94864
     // in order to properly handle new CUDA 112 behavior
-    c10::cuda::GetDevice(&device);
+    // c10::cuda uses DeviceIndex instead of int
+    // https://github.com/pytorch/pytorch/pull/119142
+    c10::cuda::GetDevice(reinterpret_cast<c10::DeviceIndex*>(&device));
```

Comment:
Is this `reinterpret_cast` safe? Looks like `c10::DeviceIndex` is `int8_t`.

https://github.com/pytorch/pytorch/blob/main/c10/core/Device.h#L19

I suppose what this `reinterpret_cast` would mean is we are only using one byte out of the int `device` variable. This may work, but seems fragile.

Reply:
```suggestion
    c10::DeviceIndex device_index;
    c10::cuda::GetDevice(&device_index);
    int device = device_index; // or maybe static_cast<int>(device_index);
```

Generally try to avoid using `reintepret_cast` unless really necessary.

Reply:
Another option is to declare device as type `DeviceIndex`.

Reply:
Yeah, but an annoying thing is that you still need to have an int-type device index for `getDeviceStats`, right?

Reply:
Yes, the commit has only changed the type at some locations so far.

Reply:
> Another option is to declare device as type `DeviceIndex`.
I was referring to the same thing you said, since we need an `int` device.